src/ntt.cpp

```diff
@@ -151,4 +152,48 @@ std::vector<uint32_t> NTT::mul_vec(const std::vector<uint32_t> &a, const std::ve
 std::vector<uint32_t> NTT::conv(const std::vector<uint32_t> &a, const std::vector<uint32_t> &b) {
     std::vector<uint32_t> c = mul_vec(ntt(a, false), ntt(b, false));
     return ntt(c, true);
-}
\ No newline at end of file
+}
+
+std::vector<uint32_t> NTT::conv_and_reduce(const std::vector<uint32_t> &a, const std::vector<uint32_t> &b, uint32_t r, uint32_t s) {
+    auto call_ntt = [this](const std::vector<uint32_t> &x, bool inverse){ return ntt(x, inverse); };
```

Comment:
It might be better to replace this lambda definition with a version that uses `std::bind`, i.e.:
```
auto call_ntt = std::bind(&Ntt::ntt, this, std::placeholders::_1, std::placeholders::_2);
```

Reply:
(Adding a `using std::placeholders` somewhere would make this a bit less verbose by allowing just `_1` and `_2` instead of `std::placeholders::_1` and `std::placeholders::_2`...)